app/models/concerns/indexable.rb

```diff
@@ -18,5 +20,19 @@ module Indexable
     after_commit on: [:destroy] do
       RemoveFromIndexJob.perform_async(self.class.name, id)
     end
+
+    def set_published_at
+      unless published_at
+        if defined?(metadata) && metadata && metadata['date']
+          begin
+            update_column(:published_at, metadata['date'])
+          rescue ActiveRecord::StatementInvalid
```

Comment:
I had this method fail when the date stored in the metadata is not valid (like just a year for example), in that case I think we should default to the `created_at` date.